File: sort/c/bubblesort.c
#include "algorithms.h"

void bubblesort(int *array, size_t length) {
  for (size_t i = 0; i < length - 1; i ++) {
    for (size_t j = 0; j < length - 1 - i; j++) {
      if (array[j] > array[j + 1]) {
        swap(&array[j], &array[j + 1]);
      }
    }
  }
}

File: sort/java/BubbleSort.java
public class BubbleSort {
    public static void sort(int[] array) {
        for (int i = 0; i < array.length - 1; i++) {
            for (int j = 0; j < array.length - i - 1; j++) {
                if (array[j] > array[j + 1]) {
                    Utils.swap(array, j, j + 1);
                }
            }
        }
    }
}
